include/rocksdb/customizable.h

```diff
@@ -179,6 +190,10 @@ class Customizable : public Configurable {
   virtual const Customizable* Inner() const { return nullptr; }
 
  protected:
+  // Some classes have both a class/long name (e.g. PutOperator) and a short
+  // name (e.g. put). Classes can override this method to return a
+  // short/alternate name.  These names are used for object
+  virtual const char* ShortName() const { return ""; }
```

Comment:
This is newly added I think. Can you explain why we need another name? What is this used for? Can be confusing for users when they use Name or ShortName

Reply:
The ShortName provides a means to define an alternative name for a class.  For example, the PutOperator is known as a "PutOperator" but also via "put".  This method allows a class to define both the class name ("PutOperator") and a short/alternate/nick name ("put").

By making it part of the base class, I do not have to repeat the same code over and over again for IsInstanceOf, and can define constants associated with the class for comparison/construction.

If you have an alternate method name you prefer (NickName? AlternativeName, ByName?), I can rename it.

Reply:
It is more like a historical naming. Maybe HistoricalName would be better? But for other classes, actually, we only have the kClassName. So, if a class does not have a name before, when we migrate it to Customizable, it should only have kClassName, am I right?

Reply:
I am not adding these names to classes that did not have them already, so in that sense they are historical.  All classes should have  Name/kClassName.  Only a select few have this alternative name.

I prefer "Byname" or "NickName" to "HistoricalName", as Historical makes it sound like we will never write another class again with these alternative names.  But if you prefer Historical, I will go with that.

Reply:
So, maybe NickName. At lease, user know that they should use Name if they can.